File: EasySwapBackend/src/types/v1/artists.go
package types

// ArtistInfo 艺术家/创作者信息
type ArtistInfo struct {
	Address        string `json:"address"`         // 钱包地址
	Username       string `json:"username"`        // 用户名
	Bio            string `json:"bio"`             // 简介
	Avatar         string `json:"avatar"`          // 头像 URL
	Banner         string `json:"banner"`          // 横幅 URL
	IsVerified     bool   `json:"is_verified"`     // 是否认证
	TotalVolume    string `json:"total_volume"`    // 总交易量
	TotalSales     int64  `json:"total_sales"`     // 总销售次数
	TotalItems     int64  `json:"total_items"`     // 创作物品总数
	FloorPrice     string `json:"floor_price"`     // 地板价
	Followers      int64  `json:"followers"`       // 粉丝数
	SocialLinks    SocialLinks `json:"social_links"` // 社交媒体链接
}

// SocialLinks 社交媒体链接
type SocialLinks struct {
	Twitter  string `json:"twitter"`
	Discord  string `json:"discord"`
	Instagram string `json:"instagram"`
	Website  string `json:"website"`
}

// ArtistRanking 艺术家排名
type ArtistRanking struct {
	Rank         int     `json:"rank"`          // 排名
	Address      string  `json:"address"`       // 钱包地址
	Username     string  `json:"username"`      // 用户名
	Avatar       string  `json:"avatar"`        // 头像
	IsVerified   bool    `json:"is_verified"`   // 是否认证
	Volume       string  `json:"volume"`        // 交易量
	VolumeChange float64 `json:"volume_change"` // 交易量变化 (%)
	Sales        int64   `json:"sales"`         // 销售次数
	FloorPrice   string  `json:"floor_price"`   // 地板价
}

// ArtistsFilterParams 艺术家过滤参数
type ArtistsFilterParams struct {
	ChainID   []int  `json:"chain_ids"`   // 链 ID 列表
	Period    string `json:"period"`      // 时间范围：1h, 6h, 1d, 7d, 30d
	Category  string `json:"category"`    // 分类
	Search    string `json:"search"`      // 搜索关键词
	Page      int    `json:"page"`        // 页码
	PageSize  int    `json:"page_size"`   // 每页数量
	SortBy    string `json:"sort_by"`     // 排序字段：volume, sales, floor_price
	SortOrder string `json:"sort_order"`  // 排序顺序：asc, desc
}

// ArtistsResponse 艺术家响应
type ArtistsResponse struct {
	Result []*ArtistRanking `json:"result"`
	Total  int64            `json:"total"`
}

File: nft-market-next/lib/api/artists.ts
import { get } from './client';
import type { ArtistsFilterParams, ArtistsResponse, ArtistInfo } from '@/types';

/**
 * 获取艺术家/创作者列表
 * @param filters 过滤参数
 * @returns 艺术家列表
 */
export const getArtists = async (filters?: ArtistsFilterParams): Promise<ArtistsResponse> => {
  const params = new URLSearchParams();
  
  if (filters) {
    if (filters.chainIds) {
      params.append('chain_ids', filters.chainIds.join(','));
    }
    if (filters.period) {
      params.append('period', filters.period);
    }
    if (filters.category) {
      params.append('category', filters.category);
    }
    if (filters.search) {
      params.append('search', filters.search);
    }
    if (filters.page) {
      params.append('page', filters.page.toString());
    }
    if (filters.pageSize) {
      params.append('page_size', filters.pageSize.toString());
    }
    if (filters.sortBy) {
      params.append('sort_by', filters.sortBy);
    }
    if (filters.sortOrder) {
      params.append('sort_order', filters.sortOrder);
    }
  }
  
  const queryString = params.toString();
  return get(`/artists${queryString ? `?${queryString}` : ''}`);
};

/**
 * 获取艺术家排名
 * @param period 时间范围
 * @param limit 数量限制
 * @returns 艺术家排名
 */
export const getArtistsRanking = async (period: string = '1d', limit: number = 10) => {
  return get<ArtistsResponse>(`/artists/ranking?period=${period}&limit=${limit}`);
};

/**
 * 获取指定艺术家详情
 * @param address 艺术家地址
 * @returns 艺术家详情
 */
export const getArtistDetail = async (address: string): Promise<ArtistInfo> => {
  return get(`/artists/${address}`);
};
